File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/beer-song/beer-song.js
export const recite = (initialBottlesCount, takeDownCount) => {
  const verses = [];
  
  const generateVerse = (current, next) => {
    const line1 = current === 0 
      ? `No more bottles of beer on the wall, no more bottles of beer.`
      : `${current} bottle${current !== 1 ? 's' : ''} of beer on the wall, ${current} bottle${current !== 1 ? 's' : ''} of beer.`;
    
    const line2 = current === 0
      ? `Go to the store and buy some more, 99 bottles of beer on the wall.`
      : `Take ${current === 1 ? 'it' : 'one'} down and pass it around, ${next === 0 ? 'no more' : next} bottle${next !== 1 ? 's' : ''} of beer on the wall.`;

    return [line1, line2];
  };

  for (let i = 0; i < takeDownCount; i++) {
    const current = initialBottlesCount - i;
    const next = current - 1;
    
    let verseLines = generateVerse(current, next);
    
    if (current === 0) {
      verseLines = generateVerse(0, 99);
    }

    verses.push(...verseLines);
    
    // Add empty line between verses unless it's the last one
    if (i !== takeDownCount - 1) {
      verses.push('');
    }
  }

  return verses;
};

File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/python/exercises/practice/beer-song/beer_song.py
def generate_verse(n):
    if n == 0:
        line1 = "No more bottles of beer on the wall, no more bottles of beer."
        line2 = "Go to the store and buy some more, 99 bottles of beer on the wall."
    else:
        plural = "bottle" if n == 1 else "bottles"
        line1 = f"{n} {plural} of beer on the wall, {n} {plural} of beer."
        
        if n == 1:
            action = "it"
            next_count = "no more"
            next_plural = "bottles"
        else:
            action = "one"
            next_count = n - 1
            if next_count == 0:
                next_count = "no more"
                next_plural = "bottles"
            else:
                next_plural = "bottle" if next_count == 1 else "bottles"
        
        line2 = f"Take {action} down and pass it around, {next_count} {next_plural} of beer on the wall."
    
    return f"{line1}\n{line2}"

def recite(start, take=1):
    verses = []
    for i in range(take):
        current = (start - i) % 100  # Handle wrap-around for numbers < 0
        verses.append(generate_verse(current))
    full_song = '\n\n'.join(verses)
    return full_song.split('\n')
